src/vector.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "vector.h"
+
+vector_t* vector_init(vector_t *v, unsigned int init_size) {
+	v->capacity = init_size;
+	v->size_factor = init_size;
+	v->count = 0;
+	v->items = malloc(sizeof(void *) * v->capacity);
```

Comment:
For future patches: If we want, we can make vectors contain `size_t n` data, where we can store a struct directly into the array (instead of a void*). Less double indirection,